### PR TITLE
894: Add physical card checkbox

### LINKED ISSUE #894: [Application] Add Physical Card Checkbox
**Is your feature request related to a problem? Please describe.**
As an applicant i want to be able to choose if i want the entitlementcard digital and /or physical. 

**Describe the solution you'd like**
- Implement an additional checkbox for physical card
- Save the information in the database

### PR DESCRIPTION
**Pull request by f1sh1918**
resolves #894

### CODE REVIEW

File: administration/src/mui-modules/application/forms/StepCardTypeForm.tsx [hunk 1]
```diff
@@ -105,15 +128,23 @@ const StepCardTypeForm: Form<State, Options, ValidatedInput, AdditionalProps> =
         <CustomDivider />
         <Typography>
           Die Ehrenamtskarte ist als physische Karte und als digitale Version für Ihr Smartphone oder Tablet erhältlich.
-          Hier können Sie wählen, ob Sie neben der physischen auch kostenfrei die digitale Ehrenamtskarte beantragen
-          möchten.
+          Hier können Sie auswählen welchen Kartentyp Sie beantragen möchten.
```
Comment (michael-markl) [inline administration/src/mui-modules/application/forms/StepCardTypeForm.tsx:131]: Sounds a bit as if the user would have to choose between one or the other? Maybe:
```suggestion
          Hier können Sie auswählen, welche Kartentypen Sie beantragen möchten.
```

File: backend/src/main/kotlin/app/ehrenamtskarte/backend/application/webservice/schema/create/Application.kt [hunk 2]
```diff
@@ -80,10 +81,16 @@ data class Application(
...
+                    translations = mapOf("de" to "Ich beantrage eine digitale Ehrenamtskarte"),
                     type = Type.Boolean,
                     value = wantsDigitalCard
                 ),
+                JsonField(
+                    name = "wantsPhysicalCard",
+                    translations = mapOf("de" to "Ich beantrage eine physische Ehrenamtskarte"),
+                    type = Type.Boolean,
+                    value = wantsPhysicalCard
+                ),
```
Comment (michael-markl) [inline backend/src/main/kotlin/app/ehrenamtskarte/backend/application/webservice/schema/create/Application.kt:93]: We should add an appropriate check in the `init` function, like:

```kt
    init {
        // ...
        if (!wantsPhysicalCard && !wantsDigitalCard) {
          throw IllegalArgumentException("Does not apply for a physical nor for a digital card.") 
        }
    }
``

File: administration/src/mui-modules/application/forms/StepCardTypeForm.tsx [hunk 3]
```diff
@@ -105,15 +128,23 @@ const StepCardTypeForm: Form<State, Options, ValidatedInput, AdditionalProps> =
...
         />
+        <SubForms.wantsPhysicalCard.Component
+          state={state.wantsPhysicalCard}
+          setState={useUpdateStateCallback(setState, 'wantsPhysicalCard')}
+          label='Ich beantrage eine physische Ehrenamtskarte.'
+          options={wantsPhysicalCardOptions}
+        />
+        {isInvalid && validationResult.code === CARD_TYPE_ERROR && (
+          <CardTypeAlert severity='error'>{validationResult.message}</CardTypeAlert>
+        )}
```
Comment (michael-markl) [inline administration/src/mui-modules/application/forms/StepCardTypeForm.tsx:147]: Is it necessary to introduce `validationResult.code` just for this?
Could we not simply show the error only if there is a message?

```suggestion
        {isInvalid && validationResult.message && (
          <CardTypeAlert severity='error'>{validationResult.message}</CardTypeAlert>
        )}
```

**Review by michael-markl — COMMENTED** [review]
Other than that, LGTM. (Have not tested)

File: administration/src/mui-modules/application/forms/StepCardTypeForm.tsx (hunk 3 above)
Comment (f1sh1918) [reply administration/src/mui-modules/application/forms/StepCardTypeForm.tsx:147]: sure we can. i just thought this is cleaner

**Review by michael-markl — APPROVED** [review]
(Have not tested)

**Pull request merged**